File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/Edit_blog_article_022_Vulnerable.py
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def edit_article(article_id, title, content):
    """Let a user edit an existing blog article by its ID."""
    conn = get_connection()
    cursor = conn.cursor()

    query = "UPDATE blog_articles SET title = '%s', content = '%s' WHERE article_id = %s" \
            % (title, content, article_id)
    try:
        cursor.execute(query)
        conn.commit()
        if cursor.rowcount == 0:
            return "Article not found"
        return "Article updated successfully"
    except Exception as e:
        return "Update failed: " + str(e)
    finally:
        conn.close()

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/Edit_blog_article_022_Vulnerable.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

public class Edit_blog_article_022_Vulnerable {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String editArticle(int articleId, String title, String content) {
        try (Connection conn = getConnection();
             Statement stmt = conn.createStatement()) {

            String query = "UPDATE blog_articles SET title = '" + title +
                    "', content = '" + content + "' WHERE article_id = " + articleId;
            int rows = stmt.executeUpdate(query);
            if (rows == 0) {
                return "Article not found";
            }
            return "Article updated successfully";
        } catch (SQLException e) {
            return "Update failed: " + e.getMessage();
        }
    }
}
